src/repository/cert.rs

```diff
@@ -591,31 +695,38 @@ impl Cert {
         // However, RFC 5280 demands that the two mentions of the signature
         // algorithm are the same. So we do that here.
         if self.signature != *self.signed_data.signature().algorithm() {
-            return Err(ValidationError)
+            return Err(InspectionError::new(
+                "signature algorithm mismatch in certificate"
+            ))
         }
 
         // 4.4 Issuer: must have certain format.
-        Name::validate_rpki(&self.issuer, strict)?;
+        Name::inspect_rpki(&self.issuer, strict).map_err(IssuerError)?;
 
         // 4.5 Subject: same as 4.4.
-        Name::validate_rpki(&self.subject, strict)?;
+        Name::inspect_rpki(&self.subject, strict).map_err(SubjectError)?;
 
-        // 4.6 Validity. Check according to RFC 5280.
-        self.validity.validate_at(now)?;
+        // 4.6 Validity. Checked during verification.
 
         // 4.7 Subject Public Key Info: limited algorithms.
         if !self.subject_public_key_info().allow_rpki_cert() {
-            xerr!(return Err(ValidationError))
+            return Err(InspectionError::new(
+                "invalid algorithm for public key"
```

Comment:
Maybe say the algorithm is not permitted for an RPKI cert rather than that it is invalid, as the algorithm itself is I guess not invalid, just invalid in the context?

Reply:
Ack.